### ISSUE TITLE
Serialization is degraded

### ISSUE THREAD

**Issue by featalion** [mcohen01/amazonica#195]
Hi. After amazonica upgrade from version 0.3.33 to 0.3.46 I run into serialization issue.
DynamoDB's `batch-write-items` raises exception `java.lang.RuntimeException: Failed to create an instance of com.amazonaws.services.dynamodbv2.model.PutRequest` if `:request-items` contains list of requests instead of vector, i.e.

```clojure
(let [batch (list
             {:put-request {:item {:id "XXXX", :field "YYYY"}}}
             {:put-request {:item {:id "ZZZZ", :field "AAAA"}}})]
  (batch-write-items cred
                     :return-consumed-capacity "TOTAL"
                     :request-items {"some-table" batch})
```

```
Caused by: com.amazonaws.AmazonClientException: Unable to marshall request to JSON:
Error setting :put-request: Failed to create an instance of
com.amazonaws.services.dynamodbv2.model.PutRequest
from {:item {:id "XXXX", :field "YYYY"}} due to clojure.lang.ExceptionInfo:
Error setting :item: . Perhaps the value isn't compatible with the setter?
{:property :item, :value {:id "XXXX", :field "YYYY"}}.
Make sure the data matches an existing constructor and setters..
Perhaps the value isn't compatible with the setter?

[skip stacktrace]

Caused by: java.lang.RuntimeException: Failed to create an instance of
com.amazonaws.services.dynamodbv2.model.PutRequest
from {:item {:id "XXXX", :field "YYYY"}}.
Make sure the data matches an existing constructor and setters.
```

Exceptions' messages do not reveal the issue and do not help to find it.

I explicitly convert lists to vectors and it works totally fine. But it is not very handy anyway, as most Clojure's functions, like `partition-all`, return sequences.

**Comment by mcohen01** [comment]
Sorry about that. Fix version is 0.3.48

**Comment by featalion** [comment]
Thank you!
